cli/lsp/diagnostics.rs

```diff
@@ -229,11 +229,12 @@ fn ts_json_to_diagnostics(
     .collect()
 }
 
+// None is returned if getting diagnostics was cancelled.
 pub async fn generate_ts_diagnostics(
   state_snapshot: StateSnapshot,
   diagnostic_collection: DiagnosticCollection,
   ts_server: &tsc::TsServer,
-) -> Result<DiagnosticVec, AnyError> {
+) -> Result<Option<DiagnosticVec>, AnyError> {
```

Comment:
Have you considered how this is handled when diagnostics are "cleared"?

Reply:
I guess it only happens when they have been cancelled... right?

Reply:
If they are cleared, a `Some` of an empty `DiagnosticVec` would be returned I think.